ZipKit/ZKCDTrailer.m

```diff
@@ -45,6 +45,7 @@ - (void) observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(
 }
 
 + (ZKCDTrailer *) recordWithData:(NSData *)data atOffset:(UInt64)offset {
+    if (!data) return nil;
```

Comment:
@dileep-hh This check (and the one in ZKCDTrailer64Locator.m) should be:
` if (!data || data.length < 1) return nil;`